inbm/dispatcher-agent/dispatcher/update_logger.py

```diff
@@ -187,6 +198,19 @@ def update_granular_with_sota_package_list_install(self) -> None:
             with open(GRANULAR_LOG_FILE, 'w') as f:
                 json.dump(data, f, indent=4)
 
+    def update_granular_with_log(self, log: dict) -> None:
+        """This function stores the log provided into the granular log file."""
```

Comment:
do we need to handle exceptions here in `open`?

Reply:
Added try and except statement to handle the exceptions